core/src/main/scala/com/banno/kafka/package.scala

```diff
@@ -179,30 +120,133 @@ package object kafka {
         x.value == y.value
     }
 
-  implicit object ProducerRecordBifunctor extends Bifunctor[ProducerRecord] {
-    def bimap[A, B, C, D](
+  implicit object ProducerRecordBitraverse extends Bitraverse[ProducerRecord] {
+    override def bifoldLeft[A, B, C](fab: ProducerRecord[A, B], c: C)(
+        f: (C, A) => C,
+        g: (C, B) => C,
+    ): C =
+      g(f(c, fab.key), fab.value)
+
+    override def bifoldRight[A, B, C](fab: ProducerRecord[A, B], c: Eval[C])(
+        f: (A, Eval[C]) => Eval[C],
+        g: (B, Eval[C]) => Eval[C],
+    ): Eval[C] =
+      g(fab.value, f(fab.key, c))
+
+    /** Note that since a record's key or value could be null, functions f and g
+      * should take care to handle null arguments. A null key means the sender
+      * was unable to choose a key on which to partition. A null value means the
+      * entity identified by the key should be deleted.
+      */
+    override def bitraverse[G[_]: Applicative, A, B, C, D](
         fab: ProducerRecord[A, B]
```

Comment:
`fab` seems like a poor name. It makes it seem like "the effect of A and B", but that does not seem like the right intuition. I guess the type is more instructive here than the name.